src/MongoDB.Driver.Core/Core/ConnectionPools/ExclusiveConnectionPool.cs

```diff
@@ -322,7 +321,16 @@ private async Task EnsureMinSizeAsync(CancellationToken cancellationToken)
                     // when adding in a connection, we need to open it because 
                     // the whole point of having a min pool size is to have
                     // them available and ready...
-                    await connection.OpenAsync(cancellationToken).ConfigureAwait(false);
+                    try
+                    {
+                        await connection.OpenAsync(cancellationToken).ConfigureAwait(false);
+                    }
+                    catch (MongoAuthenticationException)
```

Comment:
I think it should dispose on any exception type, not just an auth exception.

Reply:
Not sure that we can have any errors which will give unclosed sockets in this line, but I think there is no harm to add this suggestion. Done

Reply:
Any one of the handshake commands (ismaster, buildinfo, auth) could just fail for some reason (socket error, ok: 0), so I think it's the right thing to do.